integration/connector_test.go

```diff
@@ -0,0 +1,258 @@
+				WithStatus(http.StatusOK)
+			if err := suite.DittoClient.Reply(requestID, responseMsg); err != nil {
+				commandResponseCh <- fmt.Errorf("failed to send response: %v", err)
+				return
+			}
+			commandResponseCh <- nil
+		}
+	}
+	suite.DittoClient.Subscribe(dittoHandler)
+	defer suite.DittoClient.Unsubscribe(dittoHandler)
+
+	const commandPayload = "request"
+
+	commandTimeoutSeconds := suite.connectorTestCfg.CommandTimeoutMs / 1000
+	digitalTwinAPI := fmt.Sprintf("%s/inbox/messages/%s?timeout=%d", suite.featureURL, commandName, commandTimeoutSeconds)
+	// This request blocks until the command has been proessed and the response has been received.
+	// Run it in a goroutine, so we can process the digital twin API events while this is running.
+	body, err := util.SendDigitalTwinRequest(suite.Cfg, http.MethodPost, digitalTwinAPI, commandPayload)
+	require.NoError(suite.T(), err, "sending the command via REST and receiving response should work")
+	expectedResponse := responsePayload(commandPayload)
+	require.Equal(suite.T(), expectedResponse, string(body), "command response should match expected")
+
+	// Check the sending of the response from the feature to the backend
+	responseSentResult := waitWithTimeout(
+		suite.connectorTestCfg.CommandTimeoutMs, commandResponseCh, "receiving command and sending response timed out")
+	require.NoError(suite.T(),
+		responseSentResult,
+		"receiving command and sending response should succeed")
+}
+
+func waitWithTimeout(timeoutMS int, commandResponseCh chan error, timeoutErrorMessage string) error {
+	timeout := util.MillisToDuration(timeoutMS)
+
+	var responseSentResult error
+	select {
+	case result := <-commandResponseCh:
+		responseSentResult = result
+	case <-time.After(timeout):
+		responseSentResult = errors.New(timeoutErrorMessage)
+	}
+	return responseSentResult
+}
+
+func (suite *connectorSuite) TestEvent() {
+	suite.testModify("e", "testEvent")
+}
+
+func (suite *connectorSuite) TestTelemetry() {
+	suite.testModify("t", "testTelemetry")
+}
+
+func (suite *connectorSuite) testModify(topic string, newValue string) {
+	ws, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, "cannot create a websocket connection to the backend")
+	defer ws.Close()
+
+	filter := fmt.Sprintf("like(resource:path,'/features/%s/*')", featureID)
+	err = util.SubscribeForWSMessages(suite.Cfg, ws, "START-SEND-EVENTS", filter)
+	require.NoError(suite.T(), err, "subscription for events should succeed")
+
+	thingID := model.NewNamespacedIDFrom(suite.ThingCfg.DeviceID)
+	cmd := things.NewCommand(thingID).Twin().
+		FeatureProperty(featureID, propertyName).Modify(newValue)
+
+	msg := cmd.Envelope(protocol.WithResponseRequired(false))
+
+	err = util.SendMQTTMessage(suite.Cfg, suite.MQTTClient, topic, msg)
+
+	require.NoError(suite.T(), err, "unable to send event to the backend")
+
+	featurePropertyPath := util.GetFeaturePropertyPath(featureID, propertyName)
```

Comment:
The name has to be aligned with expectedTopic variable and the definition has to be moved to the callback, on the sam elevel as expectedTopic.